tests/kernel/interrupt/CMakeLists.txt

```diff
@@ -17,3 +17,14 @@ target_sources(app PRIVATE
 
 target_sources_ifdef(CONFIG_DYNAMIC_INTERRUPTS app PRIVATE src/dynamic_isr.c)
 target_sources_ifdef(CONFIG_X86 app PRIVATE src/regular_isr.c)
+
+if (CONFIG_SHARED_INTERRUPTS)
+# note: this test is very brittle since it requires that the chosen
+# interrupt lines are unused for all of the testing platforms. Failing
+# to meet this requirement would lead to build failures due to the number
+# of clients exceeding the limit. Still, it's important to test that the
+# static shared interrupts work properly. As such, restrict it to a single
+# platform so that we can overcome this problem.
+target_sources_ifdef(CONFIG_BOARD_QEMU_CORTEX_A53 app PRIVATE src/static_shared_irq.c)
+target_sources_ifdef(CONFIG_DYNAMIC_INTERRUPTS app PRIVATE src/dynamic_shared_irq.c)
+endif()
```

Comment:
please use `platform_allow` instead of this.

Reply:
ATM I'm under the impression that making use of `testcase.yaml` would end up becoming messier/hackier than current approach since as far as I can tell there's no way to isolate a testcase using `testcase.yaml` (please correct me if I'm wrong here). Also, `static_shared_irq.c` should only be built for `qemu_cortex_a53` and its variants otherwise we might end up having build failures as seen in previous versions of this code.  As a consequence to this, `static_shared_irq.c` would at best require a `CONFIG_GIC` instead of current `CONFIG_BOARD_QEMU_CORTEX_A53`. If the problem is that we shouldn't source files based on platform then we can quickly solve this by `#ifdef`-ing the code from `static_shared_irq.c` (not sure if this is would be better though). Will see if I can find a better, `testcase.yaml`-based solution to this.

Reply:
Done. The test is still restricted to `qemu_cortex_a53` and its variants. Can also change it to all GIC-based SoCs like you suggested (tested locally and seems fine) but it will become more prone to errors if for whatever reason testing platforms using SGIs will be added. Either way is fine with me.